pkg/storage/image.go

```diff
@@ -572,7 +581,15 @@ func (svc *imageService) ResolveNames(imageName string) ([]string, error) {
 		if r == "docker.io" && !strings.ContainsRune(remainder, '/') {
 			rem = "library/" + rem
 		}
-		images = append(images, path.Join(r, rem))
+		image := path.Join(r, rem)
+		registry, err := sysregistriesv2.FindRegistry(nil, image)
+		if err != nil {
+			return nil, err
+		}
+		if registry != nil && registry.Blocked {
+			continue
+		}
```

Comment:
One more thing… the comment above the function says
> Will only return an empty slice if err != nil.

and it seems that various callers expect this to be true.  So, after the loop, if `images` ends up empty, this should return an error (“all search locations for $remainder are blocked”) or something like that.

Reply:
done